### PR TITLE
DCMAW-8389: make APIs in test logic nicer

### PR DESCRIPTION
**Pull request by ahickmann-gds**
​DCMAW-8389

### What changed
- Some minor refactoring in `clientCredentialsGrantFlow.test.ts`

Screenshot 2024-10-07 at 17 32 24

### Why did it change
- To improve readability 

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [ ] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts [hunk 1]
```diff
@@ -7,325 +7,328 @@ import { randomUUID, UUID } from "crypto";
...
+}> => {
+  const proxyApiConfig = {
+    apiName: "Proxy API",
+    api: PROXY_API_INSTANCE,
+    authorizationHeader: "x-custom-auth",
+  };
+  const privateApiConfig = {
+    apiName: "Private API",
+    api: PRIVATE_API_INSTANCE,
+    authorizationHeader: "x-custom-auth",
```
Comment (sandymay-dd) [inline backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts:23]: 👀

Comment (ahickmann-gds) [reply backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts]: I've fixed it already! 😭 🤣

File: backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts [hunk 2]
```diff
@@ -7,325 +7,328 @@ import { randomUUID, UUID } from "crypto";
...
-  "Test %s",
-  (
-    apiInstanceName: string,
-    apiInstance: AxiosInstance,
-    authorizationHeader: string,
-  ) => {
-    let clientIdAndSecret: string;
+const getApisToTest = (): Array<{
+  apiName: string;
+  api: AxiosInstance;
```
Comment (sandymay-dd) [inline backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts:12]: I think this may want to be called axiosInstance as it's not _really_ an api

Comment (ahickmann-gds) [reply backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts:12]: Good point...

Comment (ahickmann-gds) [reply backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts]: Done!

Comment (sandymay-dd) [reply backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts]: nit(take it or leave it): stylistically in the team we adopt the shorthand syntax:

`{ key: value }[]`

**Review by sandymay-dd — DISMISSED** [review]

**Review by sandymay-dd — APPROVED** [review]

**Pull request merged**